### PR TITLE
Fixes glfwSetWindowSizeCallback conformance with glfw2 standard API.

### PR DESCRIPTION
**Pull request by guillaumeblanc**
On calls to glfwSetWindowSizeCallback, calls the provided callback function, as described by glfw2 documentation: When a callback function is set, it will be called with the current window size before this function returns.
Note that glfw3 API doesn't behaves this way according to its documentation.

### CODE REVIEW

**Comment by kripken** [comment]
I'm confused. Over here

http://www.glfw.org/docs/latest/group__window.html#gaa40cd24840daa8c62f36cafc847c72b6

it says some things changed in glfw3, but not that it used to call the callback but no longer does? Are those docs not complete, is there someplace else where you saw it say it should do the call in 2?

**Comment by guillaumeblanc** [comment]
This change from glfw 2 to 3 does not seem documented indeed, but was discussed in [glfw issue 62](https://github.com/glfw/glfw/issues/62).

Original glfw 2 glfwSetWindowSizeCallback behavior is described in [glfw 2.7 API reference manual chapter 3.2.9](http://www.glfw.org/GLFWReference27.pdf#page=22).

File: src/library_glfw.js [hunk 1]
```diff
@@ -579,6 +579,11 @@ var LibraryGLFW = {
       var win = GLFW.WindowFromId(winid);
       if (!win) return;
       win.windowSizeFunc = cbfun;
+     
+#if USE_GLFW == 2
```
Comment (juj) [inline src/library_glfw.js]: Can you add a comment here explicitly describing this unintuitive behavioral difference between GLFW 2 and 3, and perhaps a link?

**Comment by juj** [comment]
Reading the comment trail, looks good to me. However this commit changes the test file `glfw.c` to match to the updated GLFW 2 behavior, but the same file is run in GLFW 3 mode, see here https://github.com/kripken/emscripten/blob/incoming/tests/test_browser.py#L1225, so I think after this PR, the test would start to fail in GLFW 3 mode?

**Comment by guillaumeblanc** [comment]
I updated the pull request with a comment.
I ran GLFW 2 and 3 browser tests with success.
Why do you say that the same file (glfw.c) is run in GFW 3 mode. USE_GLFW flag is 2 by default, right?

**Comment by guillaumeblanc** [comment]
Are you ok to merge this request? Can I help with something?

**Comment by kripken** [comment]
Yes, 2 is the default.

This looks good to me, merged. Thanks!

**Comment by guillaumeblanc** [comment]
Hi,

I'm looking at this again as the issue is still present on master branch.

It looks to me that this file, including this merge, has not been pushed to master since [Feb 9](https://github.com/kripken/emscripten/commits/master/src/library_glfw.js).

Am I corrrect ? When is the next incoming-to-master merge planned ?

Regards,
Guillaume

**Comment by kripken** [comment]
Yeah, sorry about that, it's been a while since we merged to master. It's hard to find the resources to do it these days.

@juj, maybe we should prioritize that?

**Pull request merged**